sycl/include/CL/sycl/xilinx/fpga/opt_decorate_func.hpp

```diff
@@ -38,9 +39,9 @@ namespace sycl::xilinx {
 */
 
 template <typename T>
-void dataflow(T functor) {
+void dataflow(T&& functor) {
   _ssdm_op_SpecDataflowPipeline(-1, "");
-  functor();
+  std::forward<T>(functor)();
```

Comment:
I agree about the `T&&` above but why do we need a `std::forward<T>` here since it is just to use it here and not to forward to another function?

Reply:
the operatror() of the functor is an other function.

this can can change break user code if said user code is sufficiently weird.

example https://godbolt.org/z/cvzoPx

Reply:
Ouch! This is right... I should be careful about the hidden forwarding...
Can you add a comment about why in both places?